File: beginner/pagina08/2520/2520.java
import java.util.Scanner;

public class Main {
    
    public static void main(String[] args) {
        
        Scanner in = new Scanner(System.in);
        
        while (in.hasNextLine()) {
            String[] nxm = in.nextLine().split(" ");
            int n = Integer.parseInt(nxm[0]);
            int m = Integer.parseInt(nxm[1]);
            
            int[] X = new int[2];
            int[] Y = new int[2];
            
            for (int i = 0; i < n; i++) {
                String[] linha = in.nextLine().split(" ");
                for (int j = 0; j < m; j++) {
                    if (linha[j].equals("1")) {
                        X[0] = i;
                        X[1] = j;
                    } else if (linha[j].equals("2")) {
                        Y[0] = i;
                        Y[1] = j;
                    }
                }
            }
            
            int dist = Math.abs(X[0] - Y[0]) + Math.abs(X[1] - Y[1]);
            System.out.println(dist);
        }
        
        in.close();
    }
}

File: beginner/pagina08/2520/2520.php
<?php

while (($case = fgets(STDIN)) !== false) {

    list($n, $m) = explode(" ", trim($case));

    $X = [0, 0];
    $Y = [0, 0];

    for ($i = 0; $i < (int)$n; $i++) {
        $linha = explode(" ", trim(fgets(STDIN)));
        for ($j = 0; $j < (int)$m; $j++) {
            if ($linha[$j] === "1") $X = [$i, $j];
            if ($linha[$j] === "2") $Y = [$i, $j];
        }
    }

    $dist = abs($X[0] - $Y[0]) + abs($X[1] - $Y[1]);
    echo "$dist\n";
}